packages/devextreme/playground/jquery.html

```diff
@@ -42,22 +49,171 @@
+
+                    The Sun's influence on Earth's climate
+
+                    Solar activity has an impact on our planet's climate. Long-term changes in solar activity can lead to climate anomalies, such as the Little Ice Age in the 17th century, when there was a decrease in sunspots.
+
+                    Conclusion
+
+                    The Sun plays a key role in the existence of life on Earth. Studies of this star help not only to understand its behavior, but also to predict geomagnetic disturbances that can affect technological systems. Future studies of the Sun and its environmental impact remain an important challenge for astrophysicists and geophysicists.
+                `;
+
+                const deployment = 'gpt-4o-mini';
+                const apiVersion = '2024-02-01';
+                const endpoint = 'https://public-api.devexpress.com/demo-openai';
+                const apiKey = 'DEMO';
+
+                const aIService = new AzureOpenAI({
+                    dangerouslyAllowBrowser: true,
+                    deployment,
+                    endpoint,
+                    apiVersion,
+                    apiKey,
                 });
+
+                const generateAIServiceResponse = (prompt) => {
+                    const { system, user } = prompt;
+
+                    const params = {
+                        messages: [
+                            { role: 'system', content: system },
+                            { role: 'user', content: user },
+                        ],
+                        model: deployment,
+                        max_tokens: 1000,
+                        temperature: 0.7,
+                    };
+
+                    const aIResponsePromise = aIService.chat.completions.create(params);
+
+                    return aIResponsePromise;
+                };
+
+                const aIIntegration = {
+                    getModelResponse: (prompt, type) => {
+                        return new Promise((resolve, reject) => {
+                            generateAIServiceResponse(prompt)
+                                .then((result) => {
+                                    const data = { choices: result.choices };
+                                    const response = data.choices[0].message.content;
+
+                                    resolve(response);
+                                })
+                                .catch((error) => {
+                                    reject(error);
+                                });
+                        });
+                    },
+                };
+
+                const editor = $('#editor').dxHtmlEditor({
+                    height: 400,
+                    value: princ,
+                    toolbar: {
+                        items: [
+                            {
+                                name: 'ai',
+                                options,
+                                widget: 'dxDropDownButton'
```

Comment:
When using the name 'ai', the component is supposed to figure out which widget to use, similar to the 'undo', 'redo' buttons. I.e. the user does not have to specify the widget field here.